Project/DAQ_System/Program/Src/Platform/CAN/STM/F4/bxcan_stmf4.cpp

```diff
@@ -15,7 +15,93 @@
 
 namespace platform {
 
+BxCanStmF4::BxCanStmF4(CAN_HandleTypeDef &hcan, CAN_FilterTypeDef &filter, const std::vector<uint32_t> &can_id_list)
+  : bx_can_(hcan) {
 
+	for (uint8_t i = 0; i < can_id_list.size(); i++) {
+		ConfigureFilter(filter, i, can_id_list.at(i));
+	}
 
+	HAL_CAN_Start(&bx_can_);
+}
+
+void BxCanStmF4::ConfigureFilter(CAN_FilterTypeDef &filter, uint8_t filter_bank_num, uint32_t can_id) {
+	filter.FilterIdHigh = can_id >> 13 ;
+	filter.FilterIdLow = ((can_id & 0x1FFF) <<3) | CAN_ID_EXT | CAN_RTR_DATA ;
```

Comment:
Just making a note that this probably should be documented later since it CAN(lol) be rather confusing what is going on here